source/opt/iterator.h

```diff
@@ -183,6 +201,30 @@ inline
   return UptrVectorIterator(container_, container_->begin() + index);
 }
 
+template <typename VT, bool IC>
+template <bool IsConstForMethod>
+inline
+    typename std::enable_if<!IsConstForMethod, UptrVectorIterator<VT, IC>>::type
+    UptrVectorIterator<VT, IC>::MoveBefore(UptrVector& values) {
+  const auto pos = iterator_ - container_->begin();
+  const auto origsz = container_->size();
+  container_->resize(origsz + values.size());
+  std::move_backward(container_->begin() + pos, container_->begin() + origsz,
+                     container_->end());
+  std::move(values.begin(), values.end(), container_->begin() + pos);
```

Comment:
TIL.  I can't find a reference for this instantiation of std::move, though I can guess what this does. :-)